libraries/crypto/src/hkdf.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::hmac::hmac_256;
+use super::{Hash256, HashBlockSize64Bytes};
+
+const HASH_SIZE: usize = 32;
+
+/// Computes the HKDF with empty salt and 256 bit (one block) output.
+///
+/// # Arguments
+///
+/// * `ikm` - Input keying material
+/// * `info` - Optional context and application specific information
+///
+/// This implementation is equivalent to the below hkdf, with `salt` set to the
+/// default block of zeros and the output length l as 32.
+pub fn hkdf_empty_salt_256<H>(ikm: &[u8], info: &[u8]) -> [u8; 32]
```

Comment:
Shouldn't the return value be `[u8; HASH_SIZE]` ?

Reply:
Done